fix(core): Remove unsafe type assertion from destroyNode

mrmeku opened the pull request:
This type assertion was unsafe when using an angular element
within an angular component.

## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?
Introduce a check to see if a debug node exists before destroying it.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

mrmeku committed:
fix(core): Remove unsafe type assertion from destroyNode

This type assertion was unsafe when using an angular element
within an angular component.

mrmeku commented:
@robwormald This fix is primary targeted for people using Angular elements in dev mode.

mrmeku commented:
This change is completely non risk and is less than 4 lines changed total. Can someone take a look please?

thelgevold commented:
@jasonaden Would you have time to take a look at this PR?

chainlist commented:
Can we have an update about this PR?

Would be great if this was merged.

benoitjchevalier commented:
Waiting on this as well :)

guptasanchit90 commented:
Ran into this issue today, please review and merge this for next release. I am using Angular elements with Nx (Nrwl) and sharing these components between multiple angular apps.

ai-eye commented:
Has this been sorted somewhere yet?

gituser3000 commented:
any updates?
This bug makes cypress fail in dev mode (yes we can ignore this specific console.error).
Let's make angular great again

kevin192291 commented:
> any updates?
> This bug makes cypress fail in dev mode (yes we can ignore this specific console.error).
> Let's make angular great again

I totally agree, telling management "we just ignore the error that is thrown and it will not effect anything and the customer will not question it" Does **NOT** look good.

nimaen commented:
Well, there is a conflicting file at the moment...

kevin192291 commented:
> Well, there is a conflicting file at the moment...

If it speeds things up, I can pull the branch in the morning and fix the conflict.

alfaproject commented:
It's still missing tests, so you can keep waiting I guess

btmurrell commented:
please bump priority on this very old issue. affecting us as well.

mrmeku commented:
Hey everyone, I've updated my PR to resolve the merge conflict. I will not be writing a test for this PR since all I've done is removed an unsafe type assertion and added conditional logic. 

Cheers

guisperandio commented:
Do we have any updates on this?

vsparc commented:
Can we _please, for the love of OSS,_ get this merged?

soates commented:
Desperately need this to be merged in,

CaffeinatedCodeMonkey commented:
@mrmeku I'm not sure who to ping in regards to this issue. But, so far, there's been a lot of radio silence on this PR that's about to have a birthday. Do you have any idea when this, or an alternate fix, will be in place? Or is there someone else I should contact?

singer0503 commented:
I need this to be merged in. ~~~

mrmeku commented:
@CaffeinatedCodeMonkey Sorry, I don't have any visibility into getting this PR merged. Switching to the ivy renderer will make this PR obsolete so the likelihood of it getting attention is pretty low with Angular 9 on the horizon.

Sorry I don't have a better answer. I think it just reflects how angular has shifted from being an open source first project to a google product exported to github for public consumption.

I don't think the team has the resources to review changes from non-core team members given the priorities they face having to maintain running Angular from head within google3. Any small change needs to be tested against the entire google code base which increases the cost / friction of merging/reviewing small changes such as this one. 

I know thats a bit of disheartening explanation, but hopefully it's informative.

But for anyone waiting on this being merged I wouldn't hold out hope. I would just manually patch the framework within a `postinstall` script. You can use the npm package `patch-package` to make the process easier.

luiscla27 reviewed (approved):
Fixes [25525](https://github.com/angular/angular/issues/25525)
